### ISSUE TITLE
[Experiment] Sensitivity to mislabeled result data.

### ISSUE THREAD

**Issue by sethtroisi** [tensorflow/minigo#483]
Inspiration
-----
Training examples have a label for the result of the game (z) which we use to train the value head. We set the label to {-1 or 1} representing game is a win for black or for white. I wanted to understand the impact of this label being "wrong".

We've seen the result be "wrong" in several ways.
* @brilee did an analysis  (using a simple heuristic that expected outcome changed from value changed by a large amount after move number > 400) and found ~4% of our resign have an "unexpected outcome". Manual inspection showed a small percent of games where one side fills all it's eyes and a large group is captured or when one side passes and allows a big capture. (see last image below)
* Not cleaning up a "dead" group which changes the outcome according to Taylor-Tromp. Especially true when we limited the number of moves in a game to 520 (instead of the longer 722 move limit we use now)
* Early network strongly favoring white / black and resigning very early.

I think ideally z would be a percent change of win and would varied with our resign threshold, number of moves played in the game, age of model used to produce the game, ... 

I set out to measure what affect mislabeled value has on training, specifically the value head. 

Experiment
---
* Load 10 "golden chunks" (v9 chunk 250 to 259), this is 20 million examples taken from ~600,000 selfplay games from strong v9 models. 
* When training flip the result (from white to black win and vice versa) in some percent {1, 2, 5, 10} of examples.
* Train really fast on a bunch of TPUs at a couple of different network sizes.

Results
---
TL;DR Flipping 1% doesn't have much impact, 5% and 10% has a big impact on confidence and value accuracy.

A couple of actionable results:
* The AG paper says they aim for 5% bad resign in their self play games, if this was 10% I don't think value would learn. Similarly decreasing this to 4%, 2%, 1% has diminishing returns (given that it might increase average game length 20% or more)
* I suspect that our labels will always be X percent incorrect (should be 0.92 instead of 1 or even -0.4 instead of 1), it's possible we should set z to 0.95 or what ever the resign rate for the game was.

![image](https://user-images.githubusercontent.com/10172976/46502791-d892cf80-c7dd-11e8-9f89-4e804837cabe.png)

![image](https://user-images.githubusercontent.com/10172976/46502811-e5172800-c7dd-11e8-9e75-e4f8aa4b8ea8.png)

![image](https://user-images.githubusercontent.com/10172976/46502821-e9434580-c7dd-11e8-9ee7-0bf8bd2e4270.png)

Holdout
![image](https://user-images.githubusercontent.com/10172976/46502833-f102ea00-c7dd-11e8-9000-a726eccb27a7.png)

Filling own eye when way ahead
![image](https://user-images.githubusercontent.com/10172976/46503941-4f7d9780-c7e1-11e8-9513-7dd6bca2ed71.png)

Unexpected outcomes versus generation
![image](https://user-images.githubusercontent.com/10172976/46504241-2d384980-c7e2-11e8-9d99-aff38db90d1a.png)

## Helper script
---

```
sethtroisi@sethtroisi:~/minigo$ cat fumble_analysis
WORK_DIR="gs://$USER-sandbox/model"
DATA_DIR="gs://v9-19/data/golden_chunks"

test () {
  BOARD_SIZE=19 python dual_net.py train --use_tpu --tpu_name=sethtroisi --model_dir=$WORK_DIR/fumble_analysis/$1_$2_$3 $DATA_DIR/{250..259}.tfrecord.zz --steps=30720 --iterations_per_loop=128 --summary_steps=256 --trunk_layers=$1 --conv_width=$2 --game_result_fumble_prob=$3
}

test 5 64 0
test 5 64 0.10
test 5 64 0.05
test 5 64 0.02
test 5 64 0.01

test 10 128 0
test 10 128 0.10
test 10 128 0.05
test 10 128 0.02
test 10 128 0.01

test 20 256 0
test 20 256 0.10
test 20 256 0.05
test 20 256 0.02
test 20 256 0.01

test 5  128 0
test 15 128 0
test 20 128 0
test 5  192 0
test 10 192 0
test 15 192 0
test 20 192 0
```
